diff --git a/composer.json b/composer.json
index 3b05868..91131a3 100644
--- a/composer.json
+++ b/composer.json
@@ -5,6 +5,6 @@
   "homepage": "https://github.com/bcgov/GDX-Analytics-Drupal-Snowplow",
   "license": "Apache-2.0",
   "require": {
-    "drupal/core": "^8.8 || ^9 || ^10"
+    "drupal/core": "^8.8 || ^9 || ^10 || ^11"
   }
 }
\ No newline at end of file
diff --git a/gdx_analytics_drupal_snowplow.info.yml b/gdx_analytics_drupal_snowplow.info.yml
index a4f00cb..cba04bb 100755
--- a/gdx_analytics_drupal_snowplow.info.yml
+++ b/gdx_analytics_drupal_snowplow.info.yml
@@ -2,6 +2,7 @@ name: 'GDX Analytics Drupal Snowplow'
 type: module
 description: 'Configures the Javascript tracker for the BC Government GDX Analytics Snowplow pipeline.'
 core: 8.x
-core_version_requirement: ^8 || ^9 || ^10
+core_version_requirement: ^8 || ^9 || ^10 || ^11
 package: 'Analytics'
-version: '4.1.0'
\ No newline at end of file
+version: '4.1.0'
+configure: gdx_analytics_drupal_snowplow.gdx_analytics_drupal_snowplow_settings_form
\ No newline at end of file
diff --git a/gdx_analytics_drupal_snowplow.module b/gdx_analytics_drupal_snowplow.module
index 65e97b8..641e0ae 100644
--- a/gdx_analytics_drupal_snowplow.module
+++ b/gdx_analytics_drupal_snowplow.module
@@ -57,6 +57,7 @@ function handleAdminRoutes($logger, $messenger) {
   
   // Check if configuration is incomplete and display appropriate message.
   if (isConfigurationIncomplete($logger, $messenger)) {
+    $route_name = \Drupal::routeMatch()->getRouteName();
     if ($route_name === GDX_ANALYTICS_SETTINGS_FORM_ROUTE) {
       $messenger->addWarning(GDX_ANALYTICS_WARNING_CONFIG);
     } else {
